samples/adminconfig/infrastructure.json

```diff
@@ -1,45 +1,47 @@
 {
+    "definitions": [
+        {
+            "attribute": "storage-cost",
+            "type": "numeric",
+            "units": "US Dollar per TB per month",
+            "object": "fybrikstorageaccount"
```

Comment:
I thought the idea was to separate out the units and scales so they wouldn't have to be repeated, but the instance specific information would remain in the InfraAttribute.  Thus, wouldn't it make more sense to have the object in InfraAttribute?

Reply:
The idea was to separate the instance-specific details from common attribute details. Let's take a storage cost as an example. It is a numeric value, measured in dollars per GB, varied from 0 to 100, defined in a storage account resource. 
Now, the specifics: storage cost associated with accountX costs Y. 
This way, it is easier to understand what is affected by an attribute - e.g. storage cost affects the choice of storage accounts. Otherwise, we would end up with a single attribute "cost" that is defined for storage accounts, clusters, networks, etc. For the current solver implementation, the assumption was that an attribute is defined in a unique way, and storage-cost and cluster-cost are different attributes even if they match units and scale.

Reply:
The original idea was to have an attribute per object, i.e. storage-cost and cluster-cost are different entities. However, on the second thought, we might want to have the same "cost" attribute applied to different objects instead of separating the attributes. I've made the changes in the solver interfaces to support this. @zivnevo - please take a look (GetInstanceType has been replaced by GetInstanceTypes).

Reply:
Right.  I don't think we should have separate storage-cost and cluster-cost, just cost that can be used by multiple types of attributes.